File: ABC/ABC265/scala/A/Main.scala
import java.util.Scanner

object Main {
  def main(args: Array[String]) {
    val sc = new Scanner(System.in) 
    val X = sc.nextInt
    val Y = sc.nextInt
    val N = sc.nextInt
    if (X * 3 >= Y) {
      println(Y*(N/3) + X*(N%3))
    } else {
      println(X*N)
    }
  }
}

File: ABC/ABC265/java/A/Main.java
import java.util.*;

public class Main {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int X = sc.nextInt();
    int Y = sc.nextInt();
    int N = sc.nextInt();
    if(X * 3 >= Y) {
      System.out.println((N / 3) * Y + X*(N%3));
    } else {
      System.out.println(X * N);
    }
  }
}